spec/factories/course.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :course do
+    name { "Biology 101" }
+    subject { nil }
```

Comment:
Do you want these to be nil? I think if you just say `subject` it will use the subject factory to add in a subject. Then you would be able to use these defaults instead of filling in data from the `valid_attributes` hashes. I think this would work to reduce the amount of explicit parameter passing. FactoryBot is all about these situations where you need valid attributes and you don't really care what they are precisely.

Reply:
https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#associations

Reply:
I need to be able to create the subject and user outside of the factory and pass them in because I reference them elsewhere in the file. Also, every time I would create a new subject in the course factory, which created a new user in the subject factory, it complained that I can't create multiple users with the same email. However, I was able to remove the use of valid_attributes for passing parameters.